cmd/root.go

```diff
@@ -1,61 +1,353 @@
+		return errors.New("JIRA project required")
+	}
+
+	sinceStr := rootCmdCfg.GetString("since")
+	if sinceStr == "" {
+		rootCmdCfg.Set("since", "1970-01-01")
+	}
+	var err error
+	since, err = time.Parse("2006-01-02", sinceStr)
+	if err != nil {
+		return errors.New("Since date must be in YYYY-MM-DD format")
+	}
+	log.Debug("All config variables are valid!")
+
+	return nil
+}
+
+type Field struct {
+	ID          string   `json:"id"`
+	Key         string   `json:"key"`
+	Name        string   `json:"name"`
+	Custom      bool     `json:"custom"`
+	Orderable   bool     `json:"orderable"`
+	Navigable   bool     `json:"navigable"`
+	Searchable  bool     `json:"searchable"`
+	ClauseNames []string `json:"clauseNames"`
+	Schema      struct {
+		Type     string `json:"type"`
+		System   string `json:"system,omitempty"`
+		Items    string `json:"items,omitempty"`
+		Custom   string `json:"custom,omitempty"`
+		CustomID int    `json:"customId,omitempty"`
+	} `json:"schema,omitempty"`
+}
+
+func getFieldIDs(client jira.Client) error {
+	log.Debug("Collecting field IDs.")
+	req, err := client.NewRequest("GET", "/rest/api/2/field", nil)
+	if err != nil {
+		return err
+	}
+	fields := new([]Field)
+
+	_, err = client.Do(req, fields)
+	if err != nil {
+		return err
+	}
+
+	for _, field := range *fields {
+		switch field.Name {
+		case "GitHub ID":
+			ghIDFieldID = fmt.Sprint(field.Schema.CustomID)
+		case "GitHub Number":
+			ghNumFieldID = fmt.Sprint(field.Schema.CustomID)
+		case "GitHub Labels":
+			ghLabelsFieldID = fmt.Sprint(field.Schema.CustomID)
+		case "Last Issue-Sync Update":
+			isLastUpdateFieldID = fmt.Sprint(field.Schema.CustomID)
+		}
+	}
+
+	log.Debug("All fields have been checked.")
+
+	return nil
+}
+
+func compareIssues(ghClient github.Client, jiraClient jira.Client) error {
+	log.Debug("Collecting issues")
+	ctx := context.Background()
+
+	repo := strings.Split(rootCmdCfg.GetString("repo-name"), "/")
+
+	ghIssues, _, err := ghClient.Issues.ListByRepo(ctx, repo[0], repo[1], &github.IssueListByRepoOptions{
+		Since: since,
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if len(ghIssues) == 0 {
+		log.Info("There are no GitHub issues; exiting")
+		return nil
+	}
+	log.Debug("Collected all GitHub issues")
+
+	ids := make([]string, len(ghIssues))
+	for i, v := range ghIssues {
+		ids[i] = fmt.Sprint(*v.ID)
+	}
+
+	jql := fmt.Sprintf("project=%s AND cf[%s] in (%s)",
+		rootCmdCfg.GetString("jira-project"), ghIDFieldID, strings.Join(ids, ","))
+
+	jiraIssues, _, err := jiraClient.Issue.Search(jql, nil)
+	if err != nil {
+		return err
+	}
+
+	log.Debug("Collected all JIRA issues")
+
+	for _, ghIssue := range ghIssues {
+		found := false
+		for _, jIssue := range jiraIssues {
+			id, _ := jIssue.Fields.Unknowns.Int(fmt.Sprintf("customfield_%s", ghIDFieldID))
+			if int64(*ghIssue.ID) == id {
+				found = true
+				if err := updateIssue(*ghIssue, jIssue); err != nil {
+					return err
```

Comment:
It would probably be better to log this err and keep trucking - (and the same below).